src/main/java/ru/aston/trushanina_mu/task1/OrderTicket.java

```diff
@@ -0,0 +1,32 @@
+package ru.aston.trushanina_mu.task1;
+
+import java.math.BigDecimal;
+
+public class OrderTicket extends Order {
+
+    private final Route route;
+
+    public OrderTicket(Route route) {
+        this.route = route;
+    }
+
+    public OrderTicket(BigDecimal coefficient, BigDecimal amount, User user, int id, Route route) {
+        super(coefficient, amount, user, id);
+        this.route = route;
+    }
+
+    public Route getRoute() {
+        return route;
+    }
+
+    @Override
+    public BigDecimal getDiscount() {
+        BigDecimal result = new BigDecimal("0.0");
+        switch (route) {
+            case TRAMWAY -> result = new BigDecimal("0.06");
```

Comment:
в целом работать буде но разве не нужен break в case для уменьшения перебора? И внешний result можно было впилить в default свитча
switch (ВыражениеДляВыбора) {
           case  (Значение1):
               Код1;
               break;
           case (ЗначениеN):
               КодN;
               break;
           default:
               КодВыбораПоУмолчанию;
               break;
       }

Reply:
Notice that the new syntax uses the -> operator instead of the colon we're used to with switch statements. Also, there's no break keyword: The switch expression doesn't fall through cases. https://www.baeldung.com/java-switch
С лямбдой не проходит в другой кейс без брейка